lib/backend/k8s/resources/customresource.go

```diff
@@ -302,11 +306,55 @@ func (c *customK8sResourceClient) Watch(ctx context.Context, list model.ListInte
 	if err != nil {
 		return nil, K8sErrorToCalico(err, list)
 	}
-	return newK8sWatcherConverter(ctx, c.converter.ToKVPair, k8sWatch), nil
+	toKVPair := func(r Resource) (*model.KVPair, error) {
+		r.GetObjectKind().SetGroupVersionKind(c.k8sResourceTypeMeta.GetObjectKind().GroupVersionKind())
+		return ConvertResourceToKVPair(r, c.resourceKind)
+	}
+	return newK8sWatcherConverter(ctx, toKVPair, k8sWatch), nil
 }
 
 // EnsureInitialized is a no-op since the CRD should be
 // initialized in advance.
 func (c *customK8sResourceClient) EnsureInitialized() error {
 	return nil
 }
+
+func ListInterfaceToKey(l model.ListInterface) model.Key {
```

Comment:
these should be lower case now, and to avoid ambiguity the names should either be more descriptive or make them methods attached to the struct rather than stand alone functions.

Personally I'd go with the latter.